pyintegration/tests/test_time.py

```diff
@@ -0,0 +1,236 @@
+        dt = to_datetime(f"{datetime_str} {timezone_str}")
+        self.assertEqual(59, minute_of_day(dt, TimeZone.BT))
+
+    def test_minute_of_hour(self):
+        datetime_str = "2021-12-10T23:59:59"
+        timezone_str = "CE"
+        dt = to_datetime(f"{datetime_str} {timezone_str}")
+        self.assertEqual(59, minute_of_hour(dt, TimeZone.CE))
+
+    def test_month_of_year(self):
+        datetime_str = "2021-08-10T23:59:59"
+        timezone_str = "CH"
+        dt = to_datetime(f"{datetime_str} {timezone_str}")
+        self.assertEqual(8, month_of_year(dt, TimeZone.CH))
+
+    def test_nanos_of_day(self):
+        datetime_str = "2021-12-10T00:00:01"
+        timezone_str = "CT"
+        dt = to_datetime(f"{datetime_str} {timezone_str}")
+        self.assertEqual(10 ** 9, nanos_of_day(dt, TimeZone.CT))
+
+    def test_nanos_of_second(self):
+        datetime_str = "2021-12-10T00:00:01.000000123"
+        timezone_str = "ET"
+        dt = to_datetime(f"{datetime_str} {timezone_str}")
+        self.assertEqual(123, nanos_of_second(dt, TimeZone.ET))
+
+    def test_nanos_to_millis(self):
+        dt = now()
+        ns = nanos(dt)
+        self.assertEqual(ns // 10 ** 6, nanos_to_millis(ns))
+
+    def test_nanos_to_time(self):
+        dt = now()
+        ns = nanos(dt)
+        dt1 = nanos_to_datetime(ns)
+        self.assertEqual(dt, dt1)
+
+    def test_plus_period(self):
+        period_str = "T1H"
+        period = to_period(period_str)
+
+        dt = now()
+        dt1 = plus_period(dt, period)
+        self.assertEqual(diff_nanos(dt, dt1), 60 * 60 * 10 ** 9)
+
+        period_str = "1WT1H"
+        period = to_period(period_str)
+        dt2 = plus_period(dt, period)
+        self.assertEqual(diff_nanos(dt, dt2), (7 * 24 + 1) * 60 * 60 * 10 ** 9)
+
+    def test_plus_nanos(self):
+        dt = now()
+        dt1 = plus_nanos(dt, 1)
+        self.assertEqual(1, diff_nanos(dt, dt1))
+
+    def test_second_of_day(self):
+        datetime_str = "2021-12-10T00:01:05"
+        timezone_str = "HI"
+        dt = to_datetime(f"{datetime_str} {timezone_str}")
+        self.assertEqual(65, second_of_day(dt, TimeZone.HI))
+
+    def test_second_of_minute(self):
+        datetime_str = "2021-12-10T00:01:05"
+        timezone_str = "HK"
+        dt = to_datetime(f"{datetime_str} {timezone_str}")
+        self.assertEqual(5, second_of_minute(dt, TimeZone.HK))
+
+    def test_upper_bin(self):
+        dt = now()
+        self.assertGreaterEqual(diff_nanos(dt, upper_bin(dt, 1000000, MINUTE)), 0)
+
+    def test_year(self):
+        datetime_str = "2021-12-10T00:01:05"
+        timezone_str = "IN"
+        dt = to_datetime(f"{datetime_str} {timezone_str}")
+        self.assertEqual(2021, year(dt, TimeZone.IN))
+
+    def test_year(self):
+        datetime_str = "2021-12-10T00:01:05"
+        timezone_str = "JP"
+        dt = to_datetime(f"{datetime_str} {timezone_str}")
+        self.assertEqual(21, year_of_century(dt, TimeZone.JP))
+
```

Comment:
For all of the methods that take inputs, I would test `None` inputs so that we can properly document the behavior.